app/process/handler/consensus-message/consensus-message.go

```diff
@@ -120,6 +133,70 @@ func (cmh ConsensusMessageHandler) handleEthTxMessage(m *validatorproto.TopicEth
 	return cmh.scheduler.Cancel(m.TransactionId)
 }
 
+func (cmh ConsensusMessageHandler) verifyEthTxAuthenticity(m *validatorproto.TopicEthTransactionMessage) (bool, error) {
+	tx, _, err := cmh.ethereumClient.Client.TransactionByHash(context.Background(), common.HexToHash(m.EthTxHash))
+	if err != nil {
+		cmh.logger.Warnf("[%s] - Failed to get eth transaction by hash [%s]. Error [%s].", m.TransactionId, m.EthTxHash, err)
+		return false, err
+	}
+
+	if strings.ToLower(tx.To().String()) != strings.ToLower(cmh.bridgeContractAddress) {
+		cmh.logger.Debugf("[%s] - ETH TX [%s] - Failed authenticity - Different To Address [%s].", m.TransactionId, m.EthTxHash, tx.To().String())
+		return false, nil
+	}
+
+	txMessage, signatures, err := ethhelper.DecodeBridgeMintFunction(tx.Data())
+	if err != nil {
+		return false, err
+	}
```

Comment:
`Decode` can return `ErrorInvalidMintFunctionParameters`. This would mean that here `false` and `ErrorInvalidMintFunctionParameters` will be returned as result, and in the upper logic you are logging error and returning if `err` is returned. 

I think that `ErrorInvalidMintFunctionParameters` must be interpreted as `Eth TX not authentic` and not an error.